Patch from https://github.com/open-mpi/ompi/issues/13103#issuecomment-2662258935

base/sshmem_base_open.c:34:39: error: initialization of ‘void *’ from ‘long unsigned int’ makes pointer from integer without a cast [-Wint-conversion]
   34 | void *mca_sshmem_base_start_address = UINTPTR_MAX;
      |                                       ^~~~~~~~~~~

diff --git a/oshmem/mca/sshmem/base/sshmem_base_open.c b/oshmem/mca/sshmem/base/sshmem_base_open.c
index 1f0d1eb761..06411b3852 100644
--- a/oshmem/mca/sshmem/base/sshmem_base_open.c
+++ b/oshmem/mca/sshmem/base/sshmem_base_open.c
@@ -31,7 +31,7 @@
  * globals
  */
 
-void *mca_sshmem_base_start_address = UINTPTR_MAX;
+void *mca_sshmem_base_start_address = (void*)UINTPTR_MAX;
 
 char * mca_sshmem_base_backing_file_dir = NULL;
 
@@ -49,7 +49,7 @@ mca_sshmem_base_register (mca_base_register_flag_t flags)
                                  "base",
                                  "start_address",
                                  "Specify base address for shared memory region",
-                                 MCA_BASE_VAR_TYPE_UNSIGNED_LONG_LONG,
+                                 MCA_BASE_VAR_TYPE_UNSIGNED_LONG,
                                  NULL,
                                  0,
                                  MCA_BASE_VAR_FLAG_SETTABLE,
